picard/script.py

```diff
@@ -339,6 +339,33 @@ def register_script_function(function, name=None, eval_args=True,
     )
 
 
+def script_function(name=None, eval_args=True, check_argcount=True, prefix='func_', debug=False):
+    """Decorator helper to register script functions
+
+    It calls ``register_script_function()`` and share same arguments
+    Extra optional arguments:
+        ``prefix``: define the prefix to be removed from defined function to name script function
+                    By default, ``func_foo`` will create ``foo`` script function
+        ``debug``: if set to True, print passed parameters
+
+    Exemple:
+        @script_function(eval_args=False)
+        def func_myscriptfunc():
+            ...
+    """
+    def script_function_decorator(func):
+        nonlocal name, eval_args, check_argcount, prefix, debug
```

Comment:
I haven't run the code, but I don't see a need for `nonlocal` here.

Reply:
Nor I, but python complain with UnboundLocalError without

Reply:
It's needed for `name`, because that gets manipulated after having been read. So `nonlocal name` should be sufficient.

Reply:
Yup, I simplified, and removed debug parameter.